Documentation/design-docs/object-stack-allocation.md

```diff
@@ -0,0 +1,190 @@
+# Object Stack Allocation
+
+This document describes work to enable object stack allocation in .NET Core.
+
+## Motivation
+
+In .NET instances of object types are allocated on the garbage-collected heap.
```

Comment:
nitpick.  in .net all instances are instances of object types (i.e. ints derive from object, etc. etc).  but clearly not all of these go on the heap :)

Reply:
it might be more appropriate to say "instances of non-value types".  or something to that affect.

Reply:
Fixed.